src/ps1/ps1_memory_card.c

```diff
@@ -284,4 +369,19 @@ void ps1_memory_card_enter(void) {
     {}
     mc_enter_request = mc_enter_response = 0;
     memcard_running = 1;
+    mc_pro_command = 0;
+    game_id_length = 0x10;
+    memset(received_game_id, 0, 0x10);
```

Comment:
use sizeof(received_game_id) instead of hardcoding 0x10
not sure about the purpose of setting game_id_length initially though?

Reply:
Setting it initially is kind of a hack to make sure, that the final condition of reading the payload is not fulfilled to early. Need to find a better solution for this probably